lib/StaticAnalyzer/Core/PlistDiagnostics.cpp

```diff
@@ -786,19 +789,28 @@ static ExpansionInfo getExpandedMacro(SourceLocation MacroLoc,
   llvm::SmallString<200> ExpansionBuf;
   llvm::raw_svector_ostream OS(ExpansionBuf);
   TokenPrinter Printer(OS, PP);
+  llvm::SmallPtrSet<IdentifierInfo*, 8> AlreadyProcessedTokens;
```

Comment:
I would appreciate some comments about what this is, why do we need it etc. This entire macro expansion thing is as cryptic as it gets, so one has to counterweight that with excessive amount of comments :/